website/static/js/filepage/index.js

```diff
@@ -194,6 +195,49 @@ var FileViewPage = {
                 ], '.')
             ]);
         }
+        var sharePopover =  {view: function() { return m('button.btn.btn-sm.btn-primary.file-share', {onclick: function () {
```

Comment:
Separating the component will also enable you to pass in `link` and `height` as parameters. This is an improvement because your SharePopover would no longer have to depend on the DOM (i.e. the existence the `iframe` tag).

```js
var SharePopover = {
    view: function(ctrl, params) {
        var link = params.link;
        var heigh = params.link;
    }
}

// in FileViewPage.view
var link = $('iframe').attr('src');
var height = ...
//...
m.component(SharePopover, {link: link, height: height});
```